src/esw/science.py

```diff
@@ -0,0 +1,456 @@
+        :param device: An int that is the heater device that should be changed.
+        :param enable: A boolean that is the requested heater state of that
+            device.
+        :returns: A boolean that is the success of the transaction.
+        """
+        tx_msg = f"$HEATER_CMD,{device},{int(enable)}"
+        success = self._send_msg(tx_msg)
+
+        return success
+
+    def _read_msg(self) -> str:
+        """Reads a message on the UART receive line from the STM32 chip.
+        :returns: A string that is the received message. Note that this may be
+            an empty string if there are no messages.
+        """
+        try:
+            self._uart_lock.acquire()
+            msg = self.ser.readline()
+            self._uart_lock.release()
+        except serial.SerialException:
+            if self._uart_lock.locked():
+                self._uart_lock.release()
+            rospy.logerr("Errored in _read_msg")
+        return str(msg)
+
+    def _science_thermistor_handler(self, tx_msg: str) -> None:
+        """Processes a UART message that contains data of the three
+        thermistors onboard the carousel in Celsius. We know which sensor
+        is which because we know what the site is.
+        Then publishes to the associated ros channel.
+        :param tx_msg: A string that was received from UART that contains data
+            of the three carousel temperature sensors.
+            - Format: $SCIENCE_TEMP,<TEMP_0>,<TEMP_1>,<TEMP_2>,<EXTRA_PADDING>`
+        """
+        arr = tx_msg.split(",")
+        if len(arr) < 4:
+            return
+
+        temperature_values = ScienceTemperature()
+
+        for i in range(self._num_science_thermistors):
+            temperature_values.temperatures[i] = float(arr[i + 1])
+
+        self._publisher_by_tag[arr[0][3:]].publish(temperature_values)
+
+    def _send_mosfet_msg(self, device_name: str, enable: bool) -> bool:
+        """Sends a MOSFET message on the UART transmit line to the STM32 chip.
+        :param device_name: A string that is the device that wants to be
+            changed.
+        :param enable: A boolean that is the requested device state.
+        :returns: A boolean that is the success of sent UART transaction.
+        """
+        translated_device = self._mosfet_number_by_device_name[device_name]
+        tx_msg = self._format_mosfet_msg(translated_device, enable)
+        success = self._send_msg(tx_msg)
+        return success
+
+    def _send_msg(self, tx_msg: str) -> bool:
+        """Sends the message on the UART transmit line to the STM32 chip.
+        :param tx_msg: The raw string that is to be sent to the UART transmit
+            line.
+        :returns: A boolean that is the success of sent UART transaction.
+        """
+        try:
+            tx_msg = self._add_padding(tx_msg)
+            if len(tx_msg) > self._uart_transmit_msg_len:
+                tx_msg = tx_msg[: self._uart_transmit_msg_len]
```

Comment:
Are you sure this is okay to just cut off the message? Have we ever run into an instance where this code is run?